framed.js

```diff
@@ -4,12 +4,15 @@ var clientPromise = require('./ifrau-client');
 
 var REQUEST_KEY = require('./request-key');
 
-module.exports = function getFramedJwt(scope) {
+module.exports = function getFramedJwt(scope, opts) {
 	return clientPromise
 		.then(function(Client) {
 			return new Client().connect();
 		})
 		.then(function(client) {
+			if (opts && opts.extendSession === false) {
+				return client.request(REQUEST_KEY, scope, 'X-D2L-Session:no-keep-alive');
+			}
 			return client.request(REQUEST_KEY, scope);
```

Comment:
```suggestion
			return client.request(REQUEST_KEY, scope, opts);
```

Reply:
I will confirm and circle back, but when I debugged [this block of code](https://github.com/Brightspace/ifrau/blob/9b39ddff8ecd3cdae7b532e531fd14f71a52eef3/port/requests.js#L40), it seemed that it accepted parameters as strings like `param:value`, e.g. `X-D2L-Session:no-keep-alive`, and not objects, I will debug the change and see what happens.

Reply:
Finally got a chance to get back into this. I debugged the above request method by looking into it's usage in smart-curriculum. It seems to be handled by [this code block here](https://github.com/Brightspace/ifrau/blob/9b39ddff8ecd3cdae7b532e531fd14f71a52eef3/port/requests.js#L40). 

What I saw was that the first parameter becomes the `requestType`, so in the above case would be `REQUEST_KEY` which equates to `frau-jwt-new-jwt`. Everything after that become arguments [sent here](https://github.com/Brightspace/ifrau/blob/9b39ddff8ecd3cdae7b532e531fd14f71a52eef3/port/requests.js#L59), and which in turn then become part of the [payload](https://github.com/Brightspace/ifrau/blob/9b39ddff8ecd3cdae7b532e531fd14f71a52eef3/port/port.js#L136), that is then sent as part of a [`postMessage`](https://github.com/Brightspace/ifrau/blob/9b39ddff8ecd3cdae7b532e531fd14f71a52eef3/port/port.js#L139) call.

From searching for other usages of the method above: `"client.request('frau-jwt-new-jwt"...` I found [some results](https://search.d2l.dev/search?project=Brightspace&project=Brightspace-Configs&project=BrightspaceHypermediaComponents&project=BrightspaceUI&project=BrightspaceUILabs&project=lms&project=Scratch&full=%22client.request%28%27frau-jwt-new-jwt%22&defs=&refs=&path=&hist=&type=&xrd=&nn=7&si=full&si=full) that [send the scope as an array](https://search.d2l.dev/xref/Brightspace/gbl-map-viewer/index.html?r=5350f3f2#33).

The above method is always used to send scope, so I understand that the follow-up parameter should be `X-D2L-Session:no-keep-alive` which should be interpreted to be sent properly. I am not sure about whether it should be sent as an array, i.e. `return client.request(REQUEST_KEY, [scope, 'X-D2L-Session:no-keep-alive'])` or `return client.request(REQUEST_KEY, scope, 'X-D2L-Session:no-keep-alive')`. 🤔

Please let me know if this makes sense, and how you feel the best way to proceed with this would be. Thanks!

Reply:
The arguments being passed here are passed verbatim to the local variant (https://github.com/Brightspace/frau-jwt/blob/master/host.js#L3-L9). `'X-D2L-Session:no-keep-alive'` would not make sense as the second option, and it's expecting an options object.